frontend/src/container/PublicProfile.jsx

```diff
@@ -115,59 +120,129 @@ export class PublicProfile extends React.Component {
 
   onChangeAvatarClick = () => this.setState({ displayUploadPopup: POPUP_DISPLAY_STATE.AVATAR })
 
-  onChangeAvatarSuccess = () => {
+  onChangeAvatarSuccess = () => this.onChangeImageSuccess(
+    'avatar',
+    'profileAvatarImage',
+    'hasAvatar',
+    updateUserProfileAvatarName
+  )
+
+  onChangeCoverSuccess = () => this.onChangeImageSuccess(
+    'cover',
+    'profileCoverImage',
+    'hasCover',
+    updateUserProfileCoverName
+  )
+
+  onChangeImageSuccess = (basename, nameStateKey, hasImageKey, updateNameReducer) => {
     const { props, state } = this
-    const profileAvatarName = `avatar-${Date.now()}`
+    const name = `${basename}-${Date.now()}`
     if (state.displayedUser.userId === props.user.userId) {
-      this.props.dispatch(updateUserProfileAvatarName(profileAvatarName))
+      this.props.dispatch(updateNameReducer(name))
     }
     this.setState(oldState => {
       return {
         ...oldState,
-        displayedUser: { ...oldState.displayedUser, profileAvatarName, hasAvatar: true }
+        displayedUser: { ...oldState.displayedUser, nameStateKey: name, hasImageKey: true }
       }
     })
     this.onCloseUploadPopup()
   }
 
+  onChangeCoverClick = () => this.setState({ displayUploadPopup: POPUP_DISPLAY_STATE.COVER })
+
   onCloseUploadPopup = () => this.setState({ displayUploadPopup: undefined })
 
+  getPopupUploadImage = (uploadBaseUrl, onUploadSuccess, recommendedDimensions) => {
+    const { props } = this
+    return (
+      <PopupUploadFile
+        label={props.t('Upload an image')}
+        uploadUrl={`${uploadBaseUrl}/raw/cover`}
+        httpMethod='PUT'
+        color={GLOBAL_primaryColor} // eslint-disable-line
+        handleClose={this.onCloseUploadPopup}
+        handleSuccess={onUploadSuccess}
+        allowedMimeTypes={ALLOWED_IMAGE_MIMETYPES}
+        maximumFileSize={MAXIMUM_IMAGE_SIZE}
+      >
+        <i className='fa fa-fw fa-arrows-alt' /> {props.t('Recommended dimensions:')} {recommendedDimensions}<br />
+        <i className='fa fa-fw fa-image' /> {props.t('Maximum size: {{size}}MB', { size: MAXIMUM_IMAGE_SIZE / (1024 * 1024) })}
+      </PopupUploadFile>
+    )
+  }
+
+  isProfileOfUser = () => {
+    const { props, state } = this
+    return state.displayedUser && state.displayedUser.userId === props.user.userId
+  }
+
+  getCoverImageComponent = (changeEnabled) => {
```

Comment:
This also could be a specific component instead of a function

Reply:
Done.